clang/test/SemaCXX/constructor.cpp

```diff
@@ -96,3 +96,49 @@ namespace PR38286 {
   template<typename> struct C; // expected-note {{non-type declaration found}}
   template<typename T> C<T>::~C() {} // expected-error {{identifier 'C' after '~' in destructor name does not name a type}}
 }
+
+namespace GH121706 {
```

Comment:
Can we test move/copy constructors?

Reply:
@cor3ntin I've added tests to cover these cases and additional tests to address 
https://eel.is/c++draft/class.conv.fct#1., if I've missed any cases from the spec, I'd appreciate it if you could point them out. thanks

Reply:
That wording is for conversion operators https://compiler-explorer.com/z/bE8rv8qTv

Reply:
Oh, I see! I thought you were referring to constructors, as operators are covered in a different test [conversion-function](https://github.com/llvm/llvm-project/blob/main/clang/test/SemaCXX/conversion-function.cpp
)